core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -467,10 +474,36 @@ private void applyUpdates(TableOperations underlyingOps) {
         // re-commit each update in the chain to apply it and update current
         try {
           update.commit();
-        } catch (CommitFailedException e) {
+        } catch (CommitFailedException ce) {
           // Cannot pass even with retry due to conflicting metadata changes. So, break the
           // retry-loop.
-          throw new PendingUpdateFailedException(e);
+          throw new PendingUpdateFailedException(ce);
+
+        } catch (ValidationException e) {
+
+          // if we have enabled rollback compaction required, check current snapshot
+          if (shouldRollbackCompaction) {
+            // check parent snapshot
+            Snapshot parentSnapshot = current.snapshot(current.currentSnapshot().parentId());
```

Comment:
As discussed, this is good POC and in actual PR we can tracing back the parent in a loop